src/role/role.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import RequestWithUser from '../auth/requestWithUser.interface';
+import { JwtAuthGuard } from '../auth/jwt-auth.guard';
+import { Role } from './role.enum';
+import * as jwt from 'jsonwebtoken';
```

Comment:
I suggest to use this package https://www.npmjs.com/package/jose for handling JWT because is the certified library instead of jsonwebtoken